deepchecks/tabular/checks/model_evaluation/single_dataset_performance.py

```diff
@@ -56,7 +56,10 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         results = []
         classes = dataset.classes
         label = cast(pd.Series, dataset.label_col)
-        n_samples = label.groupby(label).count()
+        if context.with_display:
+            n_samples = label.groupby(label).count()
```

Comment:
can use Counter

Reply:
if it's the with_display case then efficiency isn't that important